### PR TITLE
FIX-#3342: Update pandas to 1.3.2

### LINKED ISSUE #3342: Update pandas to 1.3.2

### PR DESCRIPTION
**Pull request by YarShev**
Signed-off-by: Igoshev, Yaroslav <yaroslav.igoshev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3342  
- [x] tests passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/groupby.py
Comment (dchigarev) [inline modin/pandas/groupby.py:224]: removing workaround for pandas-dev/pandas#42401 bug since it was fixed in 1.3.2.

Now the removed case is handled naturally

File: modin/pandas/test/test_io.py
Comment (YarShev) [inline modin/pandas/test/test_io.py:1082]: @anmyachev , have these tests become passing for cloud due to only moving to pandas 1.3.2?
test_read_csv_empty_frame
test_read_table_empty_frame
test_read_fwf_empty_frame

File: modin/pandas/test/utils.py
Comment (YarShev) [inline modin/pandas/test/utils.py:476]:
```suggestion
    df1 : pandas.DataFrame or pandas.Series
    df2 : pandas.DataFrame or pandas.Series
```
@anmyachev , don't we check this indent with doc-checker?

File: modin/pandas/test/utils.py
Comment (YarShev) [inline modin/pandas/test/utils.py:480]:
```suggestion
    AssertionError
```

File: modin/pandas/test/utils.py
Comment (anmyachev) [reply modin/pandas/test/utils.py]: We don't do doc check for this file.

File: modin/pandas/test/test_io.py
Comment (anmyachev) [reply modin/pandas/test/test_io.py:1082]: No, due to this line https://github.com/modin-project/modin/pull/3343/files#diff-3d6f698e4f654a9e6d6ffbbb9404258f8f5eb11b986a9a61480d65bd4244e0f9R140

Comment (YarShev) [reply modin/pandas/test/test_io.py:1082]: Why should this change be put in this PR then?

Comment (anmyachev) [reply modin/pandas/test/test_io.py:1082]: As this change fixes `test_read_csv_col_handling` test (which breaks when switching to a new version of `pandas`).

**Comment by dchigarev** [comment]
The problem with OmniSci dependencies has gone after I removed `channel-priority: strict` from the conda settings. Looking into the differences between the "old" pandas 1.3.1 environment and the new one built with non-strict channel priority, it seems that there are some packages in conda-forge that does not work with the new pandas. The interesting thing is that the same versions of these packages but built not for conda-forge works fine:

<details><summary>Environment differences</summary>

```
new:
c-ares                    1.17.2               h7f98852_0    conda-forge
old:
c-ares                    1.17.1               h7f98852_1    conda-forge


new:
gcc_impl_linux-64         8.4.0               he7ac559_17
old:
gcc_impl_linux-64         8.4.0               h92c5dfe_19    conda-forge


new:
google-api-core           1.31.2             pyhd8ed1ab_0    conda-forge
old:
google-api-core           1.31.1             pyhd8ed1ab_0    conda-forge


new:
google-auth               2.0.0              pyh6c4a22f_0    conda-forge
old:
google-auth               1.34.0             pyh6c4a22f_0    conda-forge


new:
greenlet                  1.1.1            py37hcd2ae1e_0    conda-forge
old:
greenlet                  1.1.0            py37hcd2ae1e_0    conda-forge


new:
gxx_impl_linux-64         8.4.0               h9ce2e92_17
old:
gxx_impl_linux-64         8.4.0               hf2bfd01_19    conda-forge


new:
libgcc-devel_linux-64     8.4.0               hd257e2f_17
old:
libgcc-devel_linux-64     8.4.0               h5d8c923_19    conda-forge


new:
libgcc-ng                 11.1.0               hc902ee8_8    conda-forge
old:
libgcc-ng                 9.3.0               h2828fa1_19    conda-forge


new:
libgomp                   11.1.0               hc902ee8_8    conda-forge
old:
libgomp                   9.3.0               h2828fa1_19    conda-forge


new:
libstdcxx-devel_linux-64  8.4.0               hf0c5c8d_17
old:
libstdcxx-devel_linux-64  8.4.0               hb016644_19    conda-forge


new:
libwebp-base              1.2.1                h7f98852_0    conda-forge
old:
libwebp-base              1.2.0                h7f98852_2    conda-forge


new:
numpy                     1.21.2           py37h31617e3_0    conda-forge
old:
numpy                     1.21.1           py37h038b26d_0    conda-forge


new:
openssl                   1.1.1k               h7f98852_1    conda-forge
old:
openssl                   1.1.1k               h7f98852_0    conda-forge


new:
pandas                    1.3.2            py37he8f5f7f_0    conda-forge
old:
pandas                    1.3.1            py37h219a48f_0    conda-forge


new:
pyyaml                    5.4.1            py37h5e8e339_1    conda-forge
old:
pyyaml                    5.4.1            py37h5e8e339_0    conda-forge


new:
sqlalchemy                1.4.23           py37h5e8e339_0    conda-forge
old:
sqlalchemy                1.4.22           py37h5e8e339_0    conda-forge
```

</details>

**Comment by YarShev** [comment]
@dchigarev, what do you think what choice would be more appropriate between removing strict channel priority or pinning packages to concrete versions?

**Comment by dchigarev** [comment]
> @dchigarev, what do you think what choice would be more appropriate between removing strict channel priority and pinning packages to concrete versions?

@YarShev
I don't really know, if you look at diffs between these [two environments](https://github.com/modin-project/modin/pull/3343#issuecomment-901447479) you'll see that most of the differing packages are not explicitly presented in the requirements, I'm not sure that we want to include them now at the deps.

In a short-term perspective removing "strict" priority could work, but in a long term we should figure out why OmniSci env isn't working with the deps from conda-forge

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks!

**Comment by YarShev** [comment]
> > @dchigarev, what do you think what choice would be more appropriate between removing strict channel priority and pinning packages to concrete versions?
> 
> @YarShev
> I don't really know, if you look at diffs between these [two environments](https://github.com/modin-project/modin/pull/3343#issuecomment-901447479) you'll see that most of the differing packages are not explicitly presented in the requirements, I'm not sure that we want to include them now at the deps.
> 
> In a short-term perspective removing "strict" priority could work, but in a long term we should figure out why OmniSci env isn't working with the deps from conda-forge

@dchigarev , I am okay with this. @maksimbo1 , can you take a look at this issue to fix it in near future?

**Comment by devin-petersohn** [comment]
Waiting on approval from @modin-project/modin-omnisci

**Review by ienkovich — APPROVED** [review]

**Comment by dchigarev** [comment]
@YarShev can we merge this since it got all of the required approvals?

**Comment by YarShev** [comment]
@dchigarev , yes, we can.

**Comment by maksimbo1** [comment]
> @maksimbo1 , can you take a look at this issue to fix it in near future?

Okay, I'll dig into it. Looks like we "forget" some deps.

**Pull request merged**
